src/soroban.js

```diff
@@ -0,0 +1,74 @@
+import BigNumber from 'bignumber.js';
+
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * format token amount to its ideal specific number of decimals
+   *
+   * @param {BigNumber} amount - the token amount you want to display
+   * @param {number} decimals
+   * @returns {string}
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = amount.shiftedBy(-decimals).toFixed(decimals).toString();
+
+      // Trim trailing zeros
+      while (formatted[formatted.length - 1] === '0') {
+        formatted = formatted.substring(0, formatted.length - 1);
+      }
+
+      if (formatted.endsWith('.')) {
+        formatted = formatted.substring(0, formatted.length - 1);
+      }
+    }
+
+    return formatted;
+  }
+
+  /**
+   * parse token amount to use it on smart contract
+   *
+   * @param {string} value - the token amount you want to use it on smart contract
+   * @param {number} decimals
+   * @returns {string}
+   */
+  static parseTokenAmount(value, decimals) {
+    const comps = value.split('.');
+
+    let whole = comps[0];
+    let fraction = comps[1];
```

Comment:
```js
const [ whole, fraction ] = value.split('.').slice(2);
```
I think this should work?

Reply:
that works but without the slice

Reply:
yep that was a safety check in case it was > 2 items

Reply:
we should have an assertion that `.` is in the string, then: this code :boom:s if you just pass a whole number, right?

Reply:
it doesn't work though
```
value = '10.01'
'10.01'
const [ whole, fraction ] = value.split('.').slice(2);
undefined
whole
undefined
```

It doesn't actually blow up on whole numbers, the fractional piece that handles the whole number case is below this block.

Reply:
```
if (fraction === "") {
    fraction = "0";
  }
```

Reply:
could do `const [ whole, fraction ] = value.split('.').slice(0, 2);` but this doesn't need any extra checks as is.

Reply:
oh wtf :facepalm: that's what I meant re: `slice(0, 2)`

Reply:
figured as much 😄

Reply:
sorry to bring this one up again but wondering if we should be throwing if 
```
const [ whole, fraction, ..rest ] = value.split('.').slice();
if (rest.length) {.. throw?...}
```

Reply:
Yeah for sure, that's just flat-out an invalid decimal